### PR TITLE
Use Bank trasnparency API for opensource page

### PR DESCRIPTION
**Pull request by exu3**
Untested.

### CODE REVIEW

**Comment by lachlanjc** [comment]
#### 📝 Changed routes:


| [**/opensource**](https://site-gb8den1ac.hackclub.dev/opensource) |
|:-:|
|<a href="https://site-gb8den1ac.hackclub.dev/opensource">Screenshot of /opensource</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Ncngi1pQMmPw3cx0CprVkKjUJap0ob6044atXu9j0ooPY9KniQFoWNkYygkDic_f-VB1StUn8zjFCKIo9t8grg.png">(view full size)</a> |







Commit bd753b2855171e15445f0d271010a6ab5043b541 (https://site-gb8den1ac.hackclub.dev).

File: pages/opensource.js [hunk 1]
```diff
@@ -105,38 +105,20 @@ const Page = ({ repos }) => (
...
+              key={account.id}
+              name={account.name}
+              url={`https://bank.hackclub.com/${account.slug}`}
+            />
+          ))}
         <BankProject name="HQ" url={`https://bank.hackclub.com/hq/`} />
         <BankProject
           name="Endowment"
           url={`https://bank.hackclub.com/endowment`}
         />
```
Comment (sampoder) [inline pages/opensource.js]:
```suggestion
```

File: pages/opensource.js [hunk 2]
```diff
@@ -105,38 +105,15 @@ const Page = ({ repos }) => (
...
-          url="https://bank.hackclub.com/community-team"
-        />
-        <BankProject
-          name="Wild Wild West"
-          url="https://bank.hackclub.com/wild-wild-west"
-        />
-        <BankProject name="Assemble" url="https://bank.hackclub.com/assemble" />
-        <BankProject name="Epoch" url="https://bank.hackclub.com/epoch" />
+        {transparentAccounts
+          .filter(account => account.category === 'hack club hq')
```
Comment (garyhtou) [inline pages/opensource.js]:
```suggestion
          .filter(account => account.category === 'hack_club_hq')
```

Comment (cfanoulis) [inline pages/opensource.js]: other than what gary gives here, maybe move this to `getStaticProps`?

File: pages/opensource.js [hunk 3]
```diff
@@ -260,5 +237,10 @@ export async function getStaticProps() {
   const repos = await octokit.paginate('GET /orgs/{org}/repos', {
     org: 'hackclub'
   })
-  return { props: { repos }, revalidate: 30 }
+
+  const transparentAccounts = await fetch(
+    'https://bank.hackclub.com/api/v3/organizations'
+  ).then(res => res.json())
+
+  return { props: { repos, transparentAccounts }, revalidate: 30 }
```
Comment (cfanoulis) [inline pages/opensource.js:244]: hq doesnt generate projects on a half-minute basis so raising this is a good idea (2hours?)
```suggestion
  return { props: { repos, transparentAccounts }, revalidate: 60 * 60 * 2} // 2 hours
```

**Review by cfanoulis — CHANGES_REQUESTED** [review]
in addition to gary's comment,

File: pages/opensource.js [hunk 4]
```diff
@@ -105,38 +105,15 @@ const Page = ({ repos }) => (
...
-          url="https://bank.hackclub.com/community-team"
-        />
-        <BankProject
-          name="Wild Wild West"
-          url="https://bank.hackclub.com/wild-wild-west"
-        />
-        <BankProject name="Assemble" url="https://bank.hackclub.com/assemble" />
-        <BankProject name="Epoch" url="https://bank.hackclub.com/epoch" />
+        {transparentAccounts
+          .filter(account => account.category === 'hack_club_hq')
```
Comment (sampoder) [inline pages/opensource.js]:
```suggestion
          .filter(account => account.category === 'hack club hq')
```

I made this change earlier but it seems to have been reverted, the page doesn't work without the spaces left in.

File: pages/opensource.js (hunk 3 above)
Comment (sampoder) [reply pages/opensource.js:244]: There's no harm in 30 second revalidation, I'd prefer we keep it to 30 seconds

File: pages/opensource.js (hunk 4 above)
Comment (exu3) [reply pages/opensource.js]: Oops I think this is actually a Bank bug; the value for the category field is supposed to be in snake case.

File: pages/opensource.js [hunk 5]
```diff
@@ -260,5 +237,10 @@ export async function getStaticProps() {
   const repos = await octokit.paginate('GET /orgs/{org}/repos', {
     org: 'hackclub'
   })
-  return { props: { repos }, revalidate: 30 }
+
+  const transparentAccounts = await fetch(
+    'https://bank.hackclub.com/api/v3/organizations'
+  ).then(res => res.json())
```
Comment (sampoder) [inline pages/opensource.js]:
```suggestion
  const transparentAccounts = (await fetch(
    'https://bank.hackclub.com/api/v3/organizations'
  ).then(res => res.json())).filter(account => account.category === 'hack club hq')
```

File: pages/opensource.js (hunk 2 above)
Comment (sampoder) [inline pages/opensource.js]:
```suggestion
```

File: pages/opensource.js (hunk 4 above)
Comment (sampoder) [reply pages/opensource.js]: Interesting, let's leave it as is for now I guess and change it later down the line, it's working great!

**Pull request merged**